### PR TITLE
[ConstantOfShape] Use builder based interface to generate Krnl loops

### PR DESCRIPTION
**Pull request by whitneywhtsang**
Signed-off-by: Whitney Tsang <whitneyt@ca.ibm.com>

### CODE REVIEW

File: src/Conversion/ONNXToKrnl/Tensor/ConstantOfShape.cpp [hunk 1]
```diff
@@ -75,21 +75,19 @@ struct ONNXConstantOfShapeOpLowering : public ConversionPattern {
     } else
       llvm_unreachable("unsupported element type");
 
-    SmallVector<Value, 4> loopIVs;
     // Create a Krnl iterate if the output is not a scalar tensor.
-    if (!hasAllScalarValues({alloc})) {
```
[the remark this answers is not in the document]
Comment (whitneywhtsang) [reply src/Conversion/ONNXToKrnl/Tensor/ConstantOfShape.cpp:80]: Added `// CHECK-NOT: krnl.iterate` to an existing test case.

File: src/Conversion/ONNXToKrnl/Tensor/ConstantOfShape.cpp [hunk 2]
```diff
@@ -75,21 +75,22 @@ struct ONNXConstantOfShapeOpLowering : public ConversionPattern {
...
-    create.krnl.store(constantVal, alloc, loopIVs);
+      IndexExprScope childScope(&rewriter, loc);
+      ValueRange loopDef = createKrnl.defineLoops(rank);
+      SmallVector<IndexExpr, 4> lbs(rank, LiteralIndexExpr(0));
+      MemRefBoundsIndexCapture allocBounds(alloc);
+      SmallVector<IndexExpr, 4> ubs;
+      allocBounds.getDimList(ubs);
+      createKrnl.iterateIE(loopDef, loopDef, lbs, ubs,
+          [&](KrnlBuilder &createKrnl, ValueRange loopInd) {
+            // Store the constant value to the output.
```
Comment (etiotto) [inline src/Conversion/ONNXToKrnl/Tensor/ConstantOfShape.cpp:89]: [nit]: the comment does not add anything useful, is obvious a store instruction is generated to store the constant value to the output from the code itself. Please remove.

**Review by etiotto — APPROVED** [review]
LGTM

**Comment by whitneywhtsang** [comment]
@jenkins-droid test this please

**Pull request merged**